### PR TITLE
Rchain 3176: use private key file instead of private key in command line arguments for deploy

### PR DESCRIPTION
**Pull request by odeac**
## Overview
Fix the security issue of passing the private key as a command line argument to subcommand deploy

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3176

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by odeac** [comment]
bors try

**Comment by odeac** [comment]
bors try

**Comment by odeac** [comment]
bors try

**Comment by odeac** [comment]
bors try

File: casper/src/main/scala/coop/rchain/casper/util/comm/DeployRuntime.scala [hunk 1]
```diff
@@ -72,32 +73,62 @@ object DeployRuntime {
...
+        signedData = maybePrivateKey.fold(d)(SignDeployment.sign(_, d))
+
+        response <- DeployService[F].deploy(signedData)
+      } yield response.map(r => s"Response: $r")
+
+    gracefulExit(
+      readInputFiles
+        .flatMap {
+          case Left(ex) =>
+            Sync[F].delay(Left(Seq(s"Error with given file: \n${ex.getMessage}")))
```
Comment (sebtomba) [inline casper/src/main/scala/coop/rchain/casper/util/comm/DeployRuntime.scala]: This is the UI for the end users and we need to provide good error messages. Please create the whole error message in `readInputFiles`. One for the code file, one for the private key file and one for the parsing of the private key from base16.

File: casper/src/main/scala/coop/rchain/casper/util/comm/DeployRuntime.scala [hunk 2]
```diff
@@ -72,32 +73,62 @@ object DeployRuntime {
...
-              .withTimestamp(timestamp)
-
-            signedData = maybePrivateKey.fold(d)(SignDeployment.sign(_, d))
-
-            response <- DeployService[F].deploy(signedData)
-          } yield response.map(r => s"Response: $r")
-      }
+            code <- Try(Source.fromFile(file).mkString).toEither
+            maybePrivateKeyBase16 <- maybePrivateKeyFile
+                                      .traverse(f => Try(Source.fromFile(f).mkString).toEither)
```
Comment (sebtomba) [inline casper/src/main/scala/coop/rchain/casper/util/comm/DeployRuntime.scala]: `Source` needs to be closed (code and key file). Please use `Resources.withResource()` from `coop.rchain.shared`

**Pull request closed without merging**